rpcserver.go

```diff
@@ -1330,43 +1377,87 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 func (r *rpcServer) WalletBalance(ctx context.Context,
 	in *lnrpc.WalletBalanceRequest) (*lnrpc.WalletBalanceResponse, error) {
 
-	// Get total balance, from txs that have >= 0 confirmations.
-	totalBal, err := r.server.cc.wallet.ConfirmedBalance(0)
-	if err != nil {
-		return nil, err
-	}
+	var chains []*lnrpc.ChainWalletBalance
 
-	// Get confirmed balance, from txs that have >= 1 confirmations.
-	confirmedBal, err := r.server.cc.wallet.ConfirmedBalance(1)
-	if err != nil {
-		return nil, err
-	}
+	for _ , chain := range registeredChains.ActiveChains() {
+
+		cc,ok := registeredChains.LookupChain(chain)
+		if !ok {
+			return nil, fmt.Errorf("unable to get chain control for: %v", chain.String())
+		}
 
-	// Get unconfirmed balance, from txs with 0 confirmations.
-	unconfirmedBal := totalBal - confirmedBal
+		// Get total balance, from txs that have >= 0 confirmations.
+		totalBal, err := cc.wallet.ConfirmedBalance(0)
+		if err != nil {
+			return nil, err
+		}
+
+		// Get confirmed balance, from txs that have >= 1 confirmations.
+		confirmedBal, err := cc.wallet.ConfirmedBalance(1)
+		if err != nil {
+			return nil, err
+		}
+
+		// Get unconfirmed balance, from txs with 0 confirmations.
+		unconfirmedBal := totalBal - confirmedBal
+
+
+		ci := &lnrpc.ChainWalletBalance{
+			Chain: chain.String(),
+			TotalBalance:       int64(totalBal),
+			ConfirmedBalance:   int64(confirmedBal),
+			UnconfirmedBalance: int64(unconfirmedBal),
+		}
+
+		chains = append(chains, ci)
+
+	}
 
-	rpcsLog.Debugf("[walletbalance] Total balance=%v", totalBal)
 
 	return &lnrpc.WalletBalanceResponse{
-		TotalBalance:       int64(totalBal),
-		ConfirmedBalance:   int64(confirmedBal),
-		UnconfirmedBalance: int64(unconfirmedBal),
+		Chains:				chains,
 	}, nil
 }
 
+func FindChainBalanceInfo (chains *[]*lnrpc.ChainBalanceInfo, chain chainCode ) (*lnrpc.ChainBalanceInfo){
```

Comment:
This function looks like it's using the same logic as `FindPeerChain`, maybe try to combine these into a single method?

Reply:
Looks the same but different